src/techui_builder/generate.py

```diff
@@ -59,16 +60,80 @@ def _get_screen_dimensions(self, file: str) -> tuple[int, int]:
 
         return (height, width)
 
+    def _get_screen_dimensions_object(
+        self, object: EmbeddedDisplay | ActionButton
```

Comment:
We shouldn't use `object` as a variable name to avoid confusion with the builtin `object` class
```suggestion
        self, widget: EmbeddedDisplay | ActionButton
```